cadence/contracts/bridge/FlowEVMBridgeTemplates.cdc

```diff
@@ -0,0 +1,80 @@
+import "FungibleToken"
+import "NonFungibleToken"
+
+import "FlowEVMBridgeUtils"
+
+/// This contract serves Cadence code from chunked templates, replacing the contract name with the name derived from
+/// given arguments - either Cadence Type or EVM contract address.
+///
+access(all) contract FlowEVMBridgeTemplates {
+    /// Canonical path for the Admin resource
+    access(all) let AdminStoragePath: StoragePath
+    /// Chunked Hex-encoded Cadence contract code, to be joined on derived contract name
+    access(self) let templateCodeChunks: {String: [String]}
+
+    /// Serves Locker contract code for a given type, deriving the contract name from the type identifier
+    access(all) fun getLockerContractCode(forType: Type): [UInt8]? {
+        if forType.isSubtype(of: Type<@{NonFungibleToken.NFT}>()) && !forType.isSubtype(of: Type<@{FungibleToken.Vault}>()) {
+            return self.getNFTLockerContractCode(forType: forType)
+        } else if !forType.isSubtype(of: Type<@{FungibleToken.Vault}>()) && forType.isSubtype(of: Type<@{FungibleToken.Vault}>()) {
+            // TODO
+            return nil
+        }
+        return nil
+    }
+
+    /// Serves bridged asset contract code for a given type, deriving the contract name from the EVM contract info
+    // TODO: Consider adding the values we would need to derive from instead of abstracting EVM calls in scope
+    // access(all) fun getBridgedAssetContractCode(forEVMContract: EVM.EVMAddress): [UInt8]? {}
+
+    access(self) fun getNFTLockerContractCode(forType: Type): [UInt8]? {
+        if let contractName: String = FlowEVMBridgeUtils.deriveLockerContractName(fromType: forType) {
+            let contraNameHex: String = String.encodeHex(contractName.utf8)
+
+            // Construct the contract code from the templated chunked contract hex
+            let code: [UInt8] = []
+            let chunks: [String] = self.templateCodeChunks["nftLocker"]!
```

Comment:
It's because the name of the contract is derived from the type of the asset, necessarily to avoid contract name conflicts since the contracts are deployed to the single bridge Address. Each chunk is code on either side of the templates `CONTRACT_NAME` separator

Reply:
Makes sense